Makefile

```diff
@@ -1,50 +1,76 @@
+VERSION = "v0.2.0"
+DOCKER_REMOTE_REPO = "federicoponzi"
 REPO_HOME := $(shell git rev-parse --show-toplevel)
 GIT_COMMIT := $(shell git rev-parse HEAD)
 GIT_BRANCH := $(shell git rev-parse --abbrev-ref HEAD)
 APP_NAME := "horust"
 
+# Common
 COMMON_DOCKER_PARAMS := --build-arg GIT_COMMIT="$(GIT_COMMIT)" --build-arg GIT_BRANCH="$(GIT_BRANCH)"
 .PHONY: help
+.DEFAULT_GOAL := help
 
 help: ## This help.
 	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z_-]+:.*?## / {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
 
-.DEFAULT_GOAL := help
-
-# DOCKER TASKS
-# Build the container
+# General docker tasks
 build: ## Build the container
-	docker build -t $(DOCKER_REPO)/$(APP_NAME):$(VERSION) $(COMMON_DOCKER_PARAMS) .
+	docker build -t $(DOCKER_REMOTE_REPO)/$(APP_NAME):$(VERSION) $(COMMON_DOCKER_PARAMS) .
 
 build-nofeatures: ## Build the container without http requests.
-	docker build -t $(DOCKER_REPO)/$(APP_NAME)_nofeatures:$(VERSION) $(COMMON_DOCKER_PARAMS) --build-arg CARGO_PARAMS="--no-default-features" .
+	docker build -t $(DOCKER_REMOTE_REPO)/$(APP_NAME)_nofeatures:$(VERSION) $(COMMON_DOCKER_PARAMS) --build-arg CARGO_PARAMS="--no-default-features" .
 
 run: ## Run container on port configured in `config.env`
 	docker run -it --rm --env HORUST_LOG=debug -v $(REPO_HOME)/examples/services/longrunning/:/etc/horust/services/ --name="$(NAME)" $(NAME):$(VERSION)
 
 run-bash: ## Run bash with horust
-	docker run -it --rm --env HORUST_LOG=debug --name="$(NAME)" $(DOCKER_REPO)/$(APP_NAME):$(VERSION) -- /bin/bash
+	docker run -it --rm --env HORUST_LOG=debug --name="$(NAME)" $(DOCKER_REMOTE_REPO)/$(APP_NAME):$(VERSION) -- /bin/bash
 
 stop: ## Stop and remove a running container
 	docker stop $(NAME)
 
-# Docker publish
+# Docker publishing tasks
 publish: build publish-latest publish-version ## publish the `{version}` ans `latest` tagged containers to ECR
 
 publish-latest: tag-latest ## publish the `latest` tagged container
-	@echo 'publish latest to $(DOCKER_REPO)'
-	docker push $(DOCKER_REPO)/$(APP_NAME):latest
+	@echo 'publish latest to $(DOCKER_REMOTE_REPO)'
+	docker push $(DOCKER_REMOTE_REPO)/$(APP_NAME):latest
 
 publish-version: ## publish the `{version}` tagged container to ECR
-	@echo 'publish $(VERSION) to $(DOCKER_REPO)'
-	docker push $(DOCKER_REPO)/$(APP_NAME):$(VERSION)
+	@echo 'publish $(VERSION) to $(DOCKER_REMOTE_REPO)'
+	docker push $(DOCKER_REMOTE_REPO)/$(APP_NAME):$(VERSION)
 
-tag-latest:
+tag-latest: ## tags the latest container with the version listed above
 	@echo 'create tag latest'
-	docker tag $(DOCKER_REPO)/$(APP_NAME):$(VERSION) $(DOCKER_REPO)/$(APP_NAME):latest
+	docker tag $(DOCKER_REMOTE_REPO)/$(APP_NAME):$(VERSION) $(DOCKER_REMOTE_REPO)/$(APP_NAME):latest
 
 version: ## output to version
-	@echo $(VERSION)
\ No newline at end of file
+	@echo ''$(VERSION)
+
+# Docker local development tasks
+## Dargo == Docker Cargo
+dargo-prep: ## This runs everything neccessary to start developing locally in a container
```

Comment:
If inside dargo you add a check for the running container instance, one can just use that instead from the first run .

Reply:
What does a "check for the running container instance" mean exactly?

Reply:
Sorry I was from the phone! I meant the following: instead of doing dargo-create-container and then calling dargo as a two step thing, you could reduce it to a single step thing in the following way.
on line 69, when a user calls `dargo test`, before calling docker exec, you can do 
```
if docker ps | grep LOCAL_DEV_CONTAINER_NAME is empty { 
docker run [...} 
}
docker exec ...
}
```
The first time you use dargo x, it will not find the running instance thus it will first spawn it and then run your command
The second time you use dargo x, it will find the running instance and just use that. So you're basically adding a check for the running instance (paying a bit of delay time for this check) but improving the ux.

Reply:
Ah, that's pretty neat, kinda like what `docker run X` does when it doesn't find `X` locally (i.e. pull it from dockerhub).
Implementing now

Reply:
Done